pkg/operation/botanist/botanist.go

```diff
@@ -111,18 +125,14 @@ func (b *Botanist) RegisterAsSeed() error {
 
 // UnregisterAsSeed unregisters a Shoot cluster as a Seed in the Garden cluster.
 func (b *Botanist) UnregisterAsSeed() error {
-	seed, err := b.K8sGardenClient.GardenClientset().GardenV1beta1().Seeds().Get(b.Shoot.Info.Name, metav1.GetOptions{})
-	if apierrors.IsNotFound(err) {
-		return nil
-	}
-	if err != nil {
-		return err
-	}
+	shootName := b.Shoot.Info.Name
 
-	if err := b.K8sGardenClient.GardenClientset().GardenV1beta1().Seeds().Delete(seed.Name, nil); err != nil && !apierrors.IsNotFound(err) {
-		return err
+	if err := b.K8sGardenClient.GardenClientset().GardenV1beta1().Seeds().Delete(shootName, nil); err != nil && !apierrors.IsNotFound(err) {
+		// Don't try to endlessly delete a Seed if it has some Shoot clusters assign to it
+		b.Logger.Errorf("Could not unregister '%s' as Seed. This error will be ignored: '%s'", shootName, err.Error())
+		return nil
 	}
-	if err := b.K8sGardenClient.DeleteSecret(seed.Spec.SecretRef.Namespace, seed.Spec.SecretRef.Name); err != nil && !apierrors.IsNotFound(err) {
+	if err := b.K8sGardenClient.DeleteSecret(common.GardenNamespace, shootName); err != nil && !apierrors.IsNotFound(err) {
```

Comment:
Isn't it safer/clearer to use the actual secret reference in the Seed resource itself? Any reason why this was changed?